src/components/InputDropdown/index.tsx

```diff
@@ -0,0 +1,353 @@
+          }}
+          onKeyDown={e => handleKeyDown(e)}
+          contentEditable={!disabled}
+        />
+        <DropdownMenu
+          show={disabled ? false : menuVisible}
+          onMouseLeave={() => setFocusIndex(-1)}
+        >
+          {optionsArray.map((o, i) => (
+            <DropdownMenu.Item
+              key={o}
+              ref={el => itemsRef.current.set(o, el)}
+              onMouseDown={e => e.preventDefault()}
+              onClick={() => {
+                handleSelectOption(o);
+                setFocusIndex(i);
+              }}
+              onMouseMove={() => {
+                setFocusIndex(i);
+              }}
+              onKeyUp={e => e.key === 'Enter' && handleSelectOption(o)}
+              $selected={multi ? value.has(o) : value === o}
+              $forceFocus={focusIndex === i}
+              $multi={multi}
+              $disableMouseFocus
+            >
+              {o}
+            </DropdownMenu.Item>
+          ))}
+        </DropdownMenu>
+      </DropdownInputContainer>
+      {error && <DropdownErrorText>{error}</DropdownErrorText>}
+    </DropdownContainer>
+  );
+}
+
+/**
+ * EXAMPLE USAGE:
+ * app/test-page/page.tsx
+ * 
+
+'use client';
+
+import { useState } from 'react';
+import InputDropdown from '@/components/InputDropdown';
+import styled from 'styled-components';
+
+const ContainerDiv = styled.div`
+  display: grid;
+  place-items: center;
+  width: 100%;
+  min-height: 100vh;
+`;
+
+const Box = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 2rem;
+  padding: 2.5rem;
+  box-shadow: 0 0 0.2rem 0.2rem rgba(0, 0, 0, 0.2);
+  border-radius: 0.5rem;
+  margin-bottom: 2rem;
+  width: 33.75rem;
+`;
+
+export default function Page() {
+  const [fruits, setFruits] = useState(new Set<string>());
+  const [writingUtensil, setWritingUtensil] = useState('');
+  const [greenGlassDoor, setGreenGlassDoor] = useState('');
+
+  return (
+    <ContainerDiv>
+      <Box>
+        <InputDropdown
+          label="Fruits"
+          placeholder="Apple"
+          multi
+          value={fruits}
+          setValue={setFruits}
+          options={
+            new Set([
+              'Apple',
+              'Banana',
+              'Cherry',
+              'Date',
+              'Elderberry',
+              'Fig',
+              'Grape',
+              'Honeydew',
+              'Imbe',
+              'Jackfruit',
+              'Kiwi',
+              'Lime',
+              'Mango',
+              'Nectarine',
+              'Orange',
+              'Pineapple',
+              'Quince',
+              'Raspberry',
+              'Strawberry',
+              'Tomato',
+              'Vanilla Bean',
+              'Watermelon',
+              'Zucchini',
+            ])
+          }
+        />
+        <InputDropdown
+          label="Writing Utensil"
+          placeholder="Pen"
+          value={writingUtensil}
+          setValue={setWritingUtensil}
```

Comment:
question for @davidqing6432:

thoughts on having the single-select dropdown have checkboxes too? lowk i feel like that might be more consistent, but idk what the design convention is